cmd/collectors/zapi/plugins/volume/volume.go

```diff
@@ -351,5 +287,75 @@ func (my *Volume) updateVolumeLabels(data *matrix.Matrix) {
 			volume.SetLabel("all_sm_healthy", strconv.FormatBool(healthy))
 		}
 
+		_, exist := my.aggrsMap[aggrUuid]
+		volume.SetLabel("isHardwareEncrypted", strconv.FormatBool(exist))
+	}
+}
+
+func (my *Volume) getDiskData() ([]string, error) {
+	var (
+		result    []*node.Node
+		diskNames []string
+		err       error
+	)
+
+	request := node.NewXmlS("disk-encrypt-get-iter")
+	//algorithm is -- Protection mode needs to be DATA or FULL
+	// Fetching rest of them and add as
+	query := request.NewChildS("query", "")
+	encryptInfoQuery := query.NewChildS("disk-encrypt-info", "")
+	encryptInfoQuery.NewChildS("protection-mode", "open|part|miss")
+
+	if result, err = collectors.InvokeZapiCall(my.client, request, my.Logger); err != nil {
+		return nil, err
+	}
+
+	if len(result) == 0 || result == nil {
+		return nil, errors.New(errors.ERR_NO_INSTANCE, "no records found")
+	}
+
+	for _, disk := range result {
+		diskName := disk.GetChildContentS("disk-name")
+		diskNames = append(diskNames, diskName)
+	}
+	return diskNames, nil
+}
+
+func (my *Volume) updateAggrMap(disks []string, aggrDiskMap map[string][]string) {
+	// Clean aggrsMap map
+	my.aggrsMap = make(map[string]string)
+
+	for _, disk := range disks {
+		aggrData := aggrDiskMap[disk]
+		my.aggrsMap[aggrData[1]] = aggrData[0]
+	}
+}
+
+func (my *Volume) getAggrDiskMapping() (map[string][]string, error) {
+	var (
+		result        []*node.Node
+		aggrsDisksMap map[string][]string
+		err           error
+	)
+
+	request := node.NewXmlS("aggr-status-get-iter")
+	// aggrsDisksMap
+	aggrsDisksMap = make(map[string][]string)
+
+	if result, err = collectors.InvokeZapiCall(my.client, request, my.Logger); err != nil {
+		return nil, err
+	}
+
+	if len(result) == 0 || result == nil {
+		return nil, errors.New(errors.ERR_NO_INSTANCE, "no records found")
+	}
+
+	for _, aggrDisk := range result {
+		aggrUuid := aggrDisk.GetChildContentS("aaggregate-uuid")
```

Comment:
typo? Does this impact any functionality?

Reply:
done